app/controllers/application_controller.rb

```diff
@@ -125,11 +125,22 @@ def convert_hyphen_to_dash
   end
 
   def formatted_errors(type, error)
-    if type == "record_invalid"
-      error.record.errors.flat_map do |attribute, errors|
-        errors.map { |error_message| {detail: "#{attribute} #{error_message}"} }
+    case type
+    when "record_invalid"
+      error.record.errors.collect do |error|
+        # "{\"errors\":[{\"source\":{\"pointer\":\"/data/attributes/id\"},\"detail\":\"Validation failed: Suggestions weight can't be blank\"}]}"
+        if error.attribute == :base
+          case error.type
+          when :"restrict_dependent_destroy.has_many", :"restrict_dependent_destroy.has_one"
+            {source: {relationship: error.options[:record]}, detail: error.full_message}
+          else
+            {detail: "Validation failed: #{error.full_message}"}
+          end
+        else
+          {source: {pointer: "/data/attributes/#{error.attribute}"}, detail: "Validation failed: #{error.full_message}"}
+        end
       end
-    elsif type == "record_not_found"
+    when "record_not_found"
       {"errors" => [{source: {pointer: "/data/attributes/id"}, detail: error.message}]}
```

Comment:
The `record_invalid` case returns an Array, but this returns a Hash. Looks like this should only return the list on the right - most callers prepend the `errors:` key.

Reply:
Hmm yeah, good catch